octopus/src/main/scala/octopus/App.scala

```diff
@@ -0,0 +1,33 @@
+package octopus
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait App[M[_]] {
```

Comment:
Let's find some name for this type class. Any proposals?

Reply:
- `AsyncAdapter` 
- `ValidationEffect`

Reply:
From the interface point of view this reminds me applicative functor   with error handling capability, what about ‘AppError’?

Or something totally not related, like ‘Octo’?